### PR TITLE
test(components): navigation and content controls coverage (#1678)

### LINKED ISSUE #1678: Coverage: components Navigation and Content Controls
## Parent

GSA/sam-design-system#1640

## What to build

Expand test coverage for navigation and content controls in `components` (side-navigation, side-toolbar, pagination, actions-menu, video-player, and truncate-text). Cover page changes, menu selections, responsive sidebars, and text truncation toggles.

## Acceptance criteria

- [ ] Spec(s) added/extended for navigation and content control components
- [ ] Meaningful increase across all coverage metrics for `components`
- [ ] Tests pass cleanly with `npm run test:components` and `npm run coverage:check`

## Blocked by

None - can start immediately

### PR DESCRIPTION
**Pull request by fpigeonjr**
## Description

Expanded unit test coverage for navigation and content controls in `components`:
- Extended `side-navigation` covering boundary and selection lookup edge cases.
- Unskipped and expanded `side-toolbar` covering responsive view/dialog transitions, cancel, and apply actions.
- Extended `pagination` testing focus-out restoration, page boundary handling, and dropdown selections.
- Created test suite for `actions-menu` testing button triggers, CDK overlay rendering, and action events.
- Unskipped and expanded `video-player` testing preload, playback controls, and DOM cleanup.
- Created test suite for `truncate-text` testing container animation and line-limiting directive with popover overlay. 437 tests passing.

## Motivation and Context

Closes #1678
Part of EPIC #1640: Raise sam-design-system test coverage to meet and exceed QASP quality gates.

## Type of Change (Select One and Apply Github Label)

- [ ] Bug fix (non-breaking change which fixes an issue) -> Apply bugfix label
- [x] New feature (non-breaking change which adds functionality) -> Apply enhancement label
- [ ] Breaking change (fix or feature that would cause existing functionality to change) -> Apply breaking label

## Screenshots (if appropriate):

N/A — unit test additions only.

## Which browsers have you tested?

- [x] Automated tests run in jsdom / Vitest

## Checklist:

- [x] I have read the [CONTRIBUTING.md](https://github.com/GSA/sam-design-system/blob/master/CONTRIBUTING.md) document.
- [x] My code passes the automated a11y lint gate.
- [ ] This code has been reviewed by another team member and passes the reviewer checklist found in [CONTRIBUTING.md](https://github.com/GSA/sam-design-system/blob/master/CONTRIBUTING.md).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests pass; the coverage floor has not regressed.
- [x] Lint baseline not raised.
- [x] My code conforms to WCAG 2.1 AA.
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/packages/components/src/lib/actions-menu/actions-menu.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      { id: 'download', text: 'Download', mode: ActionMenuMode.DISABLED },
+      { id: 'delete', text: 'Delete', mode: ActionMenuMode.HIDDEN },
+    ],
+    ...overrides,
+  });
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [SdsMenuModule, NoopAnimationsModule],
+      declarations: [SdsActionsMenuComponent, UsaIconStubComponent],
```
[the remark this answers is not in the document]
Comment (fpigeonjr) [reply libs/packages/components/src/lib/actions-menu/actions-menu.component.spec.ts]: Fixed in e709420: Removed duplicate UsaIconStubComponent declaration from testing module declarations, keeping it exported from SdsMenuModule.

File: libs/packages/components/src/lib/video-player/video-player.component.spec.ts [hunk 2]
```diff
@@ -1,20 +1,38 @@
...
+      }
+      if (!el.querySelector('progress')) {
+        const prog = document.createElement('progress');
+        el.appendChild(prog);
+      }
+    }
+  }
 }
+(globalThis as any).InitPxVideo = MockInitPxVideo;
+(window as any).InitPxVideo = MockInitPxVideo;
```
[the remark this answers is not in the document]
Comment (fpigeonjr) [reply libs/packages/components/src/lib/video-player/video-player.component.spec.ts]: Fixed in e709420: Mocked the px-video.js module import using vi.mock and updated the mock controls to be properly toggled and restored per test in jsdom.

File: libs/packages/components/src/lib/actions-menu/actions-menu.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    component.model = {
+      trigger: { type: 'plain', shadow: false },
+      actions: [],
+    };
+    expect(() => component.ngOnInit()).not.toThrow();
+  });
+
+  it('getDisabled should return true when called', () => {
+    expect(component.getDisabled(ActionMenuMode.DISABLED)).toBe(true);
+  });
```
[the remark this answers is not in the document]
Comment (fpigeonjr) [reply libs/packages/components/src/lib/actions-menu/actions-menu.component.spec.ts]: Fixed in e709420: Fixed assignment bug in getDisabled to use strict equality comparison, and added assertion in the spec verifying false is returned for SHOWN mode.

File: libs/packages/components/src/lib/pagination/pagination.component.spec.ts [hunk 4]
```diff
@@ -72,16 +80,83 @@ describe('PaginationComponent', () => {
...
+  it('triggers onSelectChange from DOM select dropdown', () => {
+    const spy = vi.spyOn(component.pageChange, 'emit');
+    const selectEl = fixture.debugElement.query(By.css('#test-select'));
+
+    selectEl.nativeElement.value = selectEl.nativeElement.options[1].value;
+    selectEl.nativeElement.dispatchEvent(new Event('change'));
+
+    expect(spy).toHaveBeenCalled();
+    expect(component.page.pageNumber).toBe(1);
+  });
```
[the remark this answers is not in the document]
Comment (fpigeonjr) [reply libs/packages/components/src/lib/pagination/pagination.component.spec.ts]: Fixed in e709420: Added assertion verifying that selecting options[1] updates component.page.pageSize to 50 in addition to pageNumber reset and event emission.

File: libs/packages/components/src/lib/video-player/video-player.component.spec.ts [hunk 5]
```diff
@@ -58,21 +77,142 @@ describe.skip('VideoPlayerComponent', () => {
...
-    element.nativeElement.setAttribute('src', component.VPConfiguration.sourceMp4);
-    expect(element.nativeElement.getAttribute('src')).toBe(component.VPConfiguration.sourceMp4);
-    element.nativeElement.setAttribute('src', component.VPConfiguration.sourceWebm);
-    expect(element.nativeElement.getAttribute('src')).toBe(component.VPConfiguration.sourceWebm);
+    if (element) {
+      element.nativeElement.setAttribute('src', component.VPConfiguration.sourceMp4);
+      expect(element.nativeElement.getAttribute('src')).toBe(component.VPConfiguration.sourceMp4);
+      element.nativeElement.setAttribute('src', component.VPConfiguration.sourceWebm);
+      expect(element.nativeElement.getAttribute('src')).toBe(component.VPConfiguration.sourceWebm);
+    }
```
[the remark this answers is not in the document]
Comment (fpigeonjr) [reply libs/packages/components/src/lib/video-player/video-player.component.spec.ts]: Fixed in e709420: Set loadVideoSource = true to render sources and assert src/type attributes produced directly from VPConfiguration.
